### ISSUE TITLE
feat: Request for Data Backup Functionality in apitable

### ISSUE THREAD

**Issue by VXenomac** [apitable/apitable#415]
Hello,

I am writing to inquire about the possibility of adding a data backup functionality in apitable. As a user, I would like to ensure that all the data within my current workspace is backed up in case of any unforeseen data loss incidents.

Could you please let me know if there are any plans to implement a backup feature in apitable? If not, I would like to request that this feature be added to the roadmap in order to provide users with a more secure and reliable data storage solution.

Thank you for your consideration.

**Comment by mr-kelly** [comment]
> Hello,
> 
> I am writing to inquire about the possibility of adding a data backup functionality in apitable. As a user, I would like to ensure that all the data within my current workspace is backed up in case of any unforeseen data loss incidents.
> 
> Could you please let me know if there are any plans to implement a backup feature in apitable? If not, I would like to request that this feature be added to the roadmap in order to provide users with a more secure and reliable data storage solution.
> 
> Thank you for your consideration.

In APITable, we suggest three backup mechanisms:

1. **Database backup**: Obviously, APITable stores data in a database (MySQL, etc.). It means that if you backup the database, then you backup the data itself.
2. **Export .apitable package**: We have developed a package format called ".apitable", you can use it to backup data. We already have this feature, but we have not documented it.
3. **Export Excel**: The simplest way to keep your data safe is to export APITable data through Excel export.

**Comment by waterxman** [comment]
How do I use the ". api table" feature？Where is the feature?

**Comment by ChambersChan** [comment]
> How do I use the ". api table" feature？Where is the feature?

This feature can be retain data struct about node file after export and import. But it still has defect now. Only table node are supported, and other file node such as mirror and dashboards are not supported.

### USE
Take local startup as an example.  You can execute get request or input 'http://`domain`/api/v1/node/exportBundle?nodeId=`fodxxx||dstxxx`' in a new tag after login success. The node can be table or folder.
File import needs to execute post request 'http://`domain`/api/v1/node/analyzeBundle', the request parameters are file and folder node ID of new position.
![image](https://user-images.githubusercontent.com/32906036/220515854-bea4e2db-4d43-4fd2-a20b-1ab843eeda2a.png)

**Comment by waterxman** [comment]
This feature can only export data of a single node. Is there any way to export all data of a folder?

**Comment by NN-Binary** [comment]
Any progress on this feature? It seems imperative to be able to backup APItable data

**Comment by horrylala** [comment]
> 



> > How do I use the ". api table" feature？Where is the feature?
> 
> This feature can be retain data struct about node file after export and import. But it still has defect now. Only table node are supported, and other file node such as mirror and dashboards are not supported.
> 
> ### USE
> Take local startup as an example. You can execute get request or input 'http://`domain`/api/v1/node/exportBundle?nodeId=`fodxxx||dstxxx`' in a new tag after login success. The node can be table or folder. File import needs to execute post request 'http://`domain`/api/v1/node/analyzeBundle', the request parameters are file and folder node ID of new position. ![image](https://user-images.githubusercontent.com/32906036/220515854-bea4e2db-4d43-4fd2-a20b-1ab843eeda2a.png)

The first mechanism - Using mysql , how should we do with this method?

**Comment by horrylala** [comment]
> 



> > Hello,
> > I am writing to inquire about the possibility of adding a data backup functionality in apitable. As a user, I would like to ensure that all the data within my current workspace is backed up in case of any unforeseen data loss incidents.
> > Could you please let me know if there are any plans to implement a backup feature in apitable? If not, I would like to request that this feature be added to the roadmap in order to provide users with a more secure and reliable data storage solution.
> > Thank you for your consideration.
> 
> In APITable, we suggest three backup mechanisms:
> 
> 1. **Database backup**: Obviously, APITable stores data in a database (MySQL, etc.). It means that if you backup the database, then you backup the data itself.
> 2. **Export .apitable package**: We have developed a package format called ".apitable", you can use it to backup data. We already have this feature, but we have not documented it.
> 3. **Export Excel**: The simplest way to keep your data safe is to export APITable data through Excel export.

The first mechanism - using database backup. How should i bakckup and restore?

**Comment by lumos0** [comment]
> > 
> 
> > > Hello,
> > > I am writing to inquire about the possibility of adding a data backup functionality in apitable. As a user, I would like to ensure that all the data within my current workspace is backed up in case of any unforeseen data loss incidents.
> > > Could you please let me know if there are any plans to implement a backup feature in apitable? If not, I would like to request that this feature be added to the roadmap in order to provide users with a more secure and reliable data storage solution.
> > > Thank you for your consideration.
> > 
> > 
> > In APITable, we suggest three backup mechanisms:
> > 
> > 1. **Database backup**: Obviously, APITable stores data in a database (MySQL, etc.). It means that if you backup the database, then you backup the data itself.
> > 2. **Export .apitable package**: We have developed a package format called ".apitable", you can use it to backup data. We already have this feature, but we have not documented it.
> > 3. **Export Excel**: The simplest way to keep your data safe is to export APITable data through Excel export.
> 
> The first mechanism - using database backup. How should i bakckup and restore?

just mysqldump and restore？
does it work for restore data to a new instance?

**Comment by NN-Binary** [comment]
DId the MySQL structure of APITable changed since the beginning? We are scared to update and just copy MySQL data because you could have changed some field structure and might break everything

**Comment by mr-kelly** [comment]
> DId the MySQL structure of APITable changed since the beginning? We are scared to update and just copy MySQL data because you could have changed some field structure and might break everything

Of course we will

**Comment by NN-Binary** [comment]
Ok, but how do I update APITable with the data?

**Comment by mr-kelly** [comment]
To ensure your data’s security, you can use our managed version available at https://aitable.ai/.

If the Enterprise edition piques your interest, feel free to [reach out for a complimentary sales demonstration](https://aitable.ai/share/shrdaGGppsfg3pjQLXALG/fomnfpJ9XBfDDz6Dkz?fldy5ZmHYGZx2=github).

**Comment by mr-kelly** [comment]
To ensure your data’s security, you can use our managed version available at https://aitable.ai/.

If you interested in the Enterprise self-hosted edition, feel free to [reach out for a complimentary sales demonstration](https://aitable.ai/share/shrdaGGppsfg3pjQLXALG/fomnfpJ9XBfDDz6Dkz?fldy5ZmHYGZx2=github).

**Comment by NN-Binary** [comment]
Yes, we are interested with the managed version, however, we would like to know if our data is safe in the current self-hosted version and can APITable be updated safely

**Comment by mr-kelly** [comment]
> Yes, we are interested with the managed version, however, we would like to know if our data is safe in the current self-hosted version and can APITable be updated safely

We use Mybatis for schema migration

**Comment by NN-Binary** [comment]
Ok, and historically, is it safe to just update in docker with the current .data folder?

**Comment by NN-Binary** [comment]
So, I've attempted an upgrade, it seems to work, but now all our scripts are broken because of this

  message: 'The API request rate for the current space is 5 queries per second (QPS). Upgrading the space allows for a higher call frequency. Contact us: sales@aitable.ai.'

This was never said prior :/ before the limit was in the file, now we need to pay to unlock it even in the community self-hosted?
